source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/WholesaleServicesQueries.cs

```diff
@@ -26,69 +28,149 @@ namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.Calculat
+            calculationTypePerGridAreas,
+            _whereClauseProvider,
+            _deltaTableOptions.Value);
 
-        return _databricksSqlWarehouseQueryExecutor
+        return await _databricksSqlWarehouseQueryExecutor
             .ExecuteStatementAsync(sqlStatement)
             .AnyAsync()
-            .AsTask();
+            .ConfigureAwait(false);
     }
 
-    protected override string CalculationIdColumnName => WholesaleResultColumnNames.CalculationId;
-
-    protected override string TimeColumnName => WholesaleResultColumnNames.Time;
-
-    protected override bool RowBelongsToNewPackage(RowData current, RowData previous)
+    protected override bool RowBelongsToNewPackage(DatabricksSqlRow current, DatabricksSqlRow previous)
     {
-        var isInDifferentCalculationPeriod = current.CalculationPeriod != previous.CalculationPeriod;
-
-        return HasDifferentColumnValues(current.Row, previous.Row) || isInDifferentCalculationPeriod;
+        return WholesaleServicesQueryStatement.ColumnsToGroupBy.Any(column => current[column] != previous[column])
+               || current[WholesaleResultColumnNames.CalculationId] != previous[WholesaleResultColumnNames.CalculationId];
     }
 
-    protected override WholesaleServices CreatePackageFromRowData(RowData rowData, List<WholesaleTimeSeriesPoint> timeSeriesPoints)
+    protected override WholesaleServices CreatePackageFromRowData(
+        DatabricksSqlRow rowData,
+        List<WholesaleTimeSeriesPoint> timeSeriesPoints)
     {
-        return WholesaleServicesFactory.Create(
-            rowData.Row,
-            timeSeriesPoints,
-            rowData.CalculationPeriod.CalculationVersion);
+        return WholesaleServicesFactory.Create(rowData, timeSeriesPoints);
     }
 
     protected override WholesaleTimeSeriesPoint CreateTimeSeriesPoint(DatabricksSqlRow row)
     {
         return WholesaleTimeSeriesPointFactory.Create(row);
     }
 
-    private static bool HasDifferentColumnValues(DatabricksSqlRow row1, DatabricksSqlRow row2)
+    private async Task<List<CalculationTypePerGridArea>> GetCalculationTypePerGridAreasAsync(
+        WholesaleServicesQueryParameters queryParameters)
+    {
+        var calculationTypePerGridAreaStatement =
+            new CalculationTypePerGridAreaStatement(_deltaTableOptions.Value, _whereClauseProvider, queryParameters);
+
+        var calculationTypePerGridAreas = await _databricksSqlWarehouseQueryExecutor
+            .ExecuteStatementAsync(calculationTypePerGridAreaStatement, Format.JsonArray)
+            .Select(d =>
+            {
+                var databricksSqlRow = new DatabricksSqlRow(d);
+                return new CalculationTypePerGridArea(
+                    databricksSqlRow[WholesaleResultColumnNames.GridArea]!,
+                    databricksSqlRow[WholesaleResultColumnNames.CalculationType]!);
+            })
+            .ToListAsync()
+            .ConfigureAwait(false);
+
+        var foo = calculationTypePerGridAreas
```

Comment:
calculationTypePerGridAreasDistinct?
```suggestion
        var calculationTypePerGridAreasDistinct?= calculationTypePerGridAreas
```